Fix newname => selector pairs

akdor1154 opened the pull request:
This should fix #279 , that is, make `select(tbl, (:newname => :existing,))` work.

One note is that it's now not possible to to select `:col => mapfn` when `!(mapFn isa Function)`. I don't think it's possible to resolve this succinctly/with dispatch (see: https://discourse.julialang.org/t/functions-and-callable-methods/2983). A workaround for users using functors like this would be `select(tbl, (mapFn.(tbl.columns.col),))`.

If the above is not acceptable, then alternatively I could special-case the selector `Pair{Symbol, Union{Symbol, Int, String, AbstractArray}}`, but this seems less general that ideal.

joshday commented on src/columns.jl:
```diff
@@ -227,10 +227,10 @@ function _colindex(fnames::Union{Tuple, AbstractArray}, col, default=nothing)
-    elseif isa(col, Pair{Symbol, <:Pair}) # recursive pairs
-        return _colindex(fnames, col[2])
-    elseif isa(col, Pair{<:Any, <:Any})
+    elseif isa(col, Pair{<:Any, <:Function}) # selector => mapfn
```
```suggestion
    elseif isa(col, Pair{<:Any, <:Base.Callable}) # selector => mapfn
```

joshday replied:
From just a cursory glance at this, we could use `Base.Callable` instead of `Function` to allow for callable types

joshday commented:
Thanks!  I'll blame GitHub's new notification system for missing this.

akdor1154 commented:
no worries haha. Added your change (thanks!), checked tests still pass, and squashed (left the overload commit separate on purpose for imo clearer history).

joshday commented:
Thanks again!
